### PR TITLE
[Projects] Add `ensure_project` to CLI commands 

### PR DESCRIPTION
**Pull request by Tankilevitch**
Contains fixes of:
https://github.com/mlrun/mlrun/pull/2092#pullrequestreview-1023166433
https://github.com/mlrun/mlrun/pull/2063#pullrequestreview-1023170144

Also added fix for release-notes generation to use `git log old_branch..new_branch` instead of `git log old_branch..HEAD`.

### CODE REVIEW

File: mlrun/__main__.py [hunk 1]
```diff
@@ -259,7 +259,7 @@ def run(
 
     if func_url or kind or image:
         if func_url:
-            runtime = func_url_to_runtime(func_url, save_project)
+            runtime = func_url_to_runtime(func_url, ensure_project)
```
Comment (Hedingber) [inline mlrun/__main__.py:262]: You're still ensuring the project only in a specific flow of `run`

File: mlrun/__main__.py [hunk 2]
```diff
@@ -561,14 +576,14 @@ def deploy(
...
-    save_project,
+    ensure_project,
 ):
     """Deploy model or function"""
     if env_file:
         mlrun.set_env_from_file(env_file)
 
     if func_url:
-        runtime = func_url_to_runtime(func_url, save_project)
+        runtime = func_url_to_runtime(func_url, ensure_project)
```
Comment (Hedingber) [inline mlrun/__main__.py:586]: Same for `deploy`

**Review by Hedingber — CHANGES_REQUESTED** [review]

File: mlrun/api/utils/projects/follower.py [hunk 3]
```diff
@@ -83,6 +83,7 @@ def create_project(
         projects_role: typing.Optional[mlrun.api.schemas.ProjectsRole] = None,
         leader_session: typing.Optional[str] = None,
         wait_for_completion: bool = True,
+        commit_before_get: bool = False,
```
Comment (quaark) [inline mlrun/api/utils/projects/follower.py:86]: Why do we default to false here?
And here is the value set to true so we indeed commit before the get?

File: automation/release_notes/generate.py [hunk 4]
```diff
@@ -57,6 +57,7 @@ def __init__(
             "Nick Brown": "ihs-nick",
             "Oded Messer": "omesser",
             "Tom Tankilevitch": "tankilevitch",
+            "Adam": "quaark",
```
Comment (quaark) [inline automation/release_notes/generate.py:60]:
```suggestion
            "Adam Melnick": "quaark",
```

**Review by quaark — CHANGES_REQUESTED** [review]

**Review by quaark — APPROVED** [review]

**Pull request merged**